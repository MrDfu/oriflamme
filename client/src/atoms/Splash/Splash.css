.splash {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  width: 60%;
  margin: 0 auto;
  color: var(--color-white);
}

.splash__image {
  height: 20vh;
  width: 20vh;
  background-image: url('../../assets/wax-seal.png');
  background-size: cover;
}

.splash__title {
  font-family: var(--font-family-gothic);
  font-size: 12rem;
}

.splash__headline {
  padding: 3rem;
  font-family: var(--font-family-regular);
  font-size: 6rem;
}

.splash__text {
  padding: 2rem;
  font-family: var(--font-family-regular);
  font-size: 2.6rem;
}

.splash__buttons {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50%;
  padding: 3rem;
  font-family: var(--font-family-gothic);
  font-size: var(--font-size-bigger);
}