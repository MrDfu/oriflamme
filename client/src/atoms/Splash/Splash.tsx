import React, {useEffect} from 'react';
import ReactModal from 'react-modal';

import './Splash.css';
import Button from '../Button/Button';

type Props={
  show:boolean;
  dismiss: ()=>void
}

type SplashModalStyles = {
  overlay: {
    zIndex: 100,
    padding: 0,
    backgroundColor: 'var(--color-blue-midnight)',
  },
  content: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    margin: '0 auto',
    border: 'none',
    backgroundColor: 'var(--color-blue-midnight)',
  }
};

export default function Splash (props:Props) {
  const { show, dismiss } = props;

 

 const splashModalStyles: SplashModalStyles={
  overlay: {
    zIndex: 100,
    padding: 0,
    backgroundColor: 'var(--color-blue-midnight)',
  },
  content: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    margin: '0 auto',
    border: 'none',
    backgroundColor: 'var(--color-blue-midnight)',
  }
};
  return (
    <ReactModal
      isOpen={show}
      onRequestClose={dismiss}
      style={splashModalStyles}
      contentLabel="Oriflamme"
    >
      <div className="splash">
        <div className="splash__title">Oriflamme</div>
        <div className="splash__headline">The King is dead. Long live the King!</div>
        <div className="splash__text">In this game for 3 - 5 players you play a powerful French family with a claim to the throne. Make it yours by laying careful plans and don't be afraid to cheat, stab, and poison your rivals to get to where you need.</div>
        <div className="splash__text">Oriflamme is produced by Studio H Games and rightly won Game of the Year in France in 2020. This is in no way endorsed by them, in fact I didn't even ask. Sorry.</div>
        <div className="splash__buttons">
          <Button
            onClick={dismiss}
            text="Play"
          />
        </div>
        <div className="splash__image"></div>
      </div>
    </ReactModal>
  );
}