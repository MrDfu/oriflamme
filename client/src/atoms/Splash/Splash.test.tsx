import { render, screen } from "@testing-library/react";
import Splash from "./Splash";
import * as React from "react";
import userEvent from "@testing-library/user-event";

describe("Splash component", () => {
  let toogle = true;
  test("should match the snapshot", () => {
    const { container } = render(
      <Splash
        show={toogle}
        dismiss={() => {
          toogle = !toogle;
        }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  test("should hide the modal when clicking the button", async () => {
    const { getByText } = render(
      <div>
        <Splash
          show={toogle}
          dismiss={() => {
            toogle = !toogle;
          }}
        />{" "}
        <div>{"Hello"}</div>
      </div>
    );
    const testButton = screen.getByRole("button", { name: /play/i });
    await userEvent.click(testButton);
    expect(getByText("Hello")).toBeTruthy();
  });
});
