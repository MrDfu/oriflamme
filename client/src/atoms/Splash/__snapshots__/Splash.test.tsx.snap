// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash component should match the snapshot 1`] = `null`;
